### DISCUSSION TITLE
Typefiles fields description

### DISCUSSION THREAD

**Discussion by mzawitowski** [SamWSoftware/serverless-auto-swagger#63] [Q&A] [answered]
Hi,

Are there any plans to add a possibility to describe fields in a *.ts files which are used to generate request and response swagger doc?

Regards,
Michal.

**Comment by bfaulk96** [comment]
I assume @mzawitowski is asking the same question as @jguerinet was in https://github.com/completecoding/serverless-auto-swagger/issues/2#issuecomment-1184683685, so I'll post my answer here:

We're using typeconv to convert TS to JSON Schema, so this is likely out of the scope of what we can accomplish in this library, although I haven't had time to look into improving types.

**Comment by jguerinet** [comment, marked as answer]
If you are indeed talking about field descriptions, there is actually a way of doing that. More info in my answer [here](https://github.com/completecoding/serverless-auto-swagger/issues/2#issuecomment-1185048853).

**Reply by bfaulk96** [reply]
I'm going to go ahead and repost [your solution](https://github.com/completecoding/serverless-auto-swagger/issues/2#issuecomment-1185048853), just so it's here to answer the question:

> I dug a bit in typeconv and you can actually do this with the following syntax:
> ```typescript
> export type Object = {
>     /** This will generate a description for the field */
>     field: string;
> };
> ```
> Here's the result:
> ![image](https://user-images.githubusercontent.com/12897328/180824095-eaaa9c42-d625-4c87-8c28-f2ee2d0a43cc.png)
> No way to remove the title field unfortunately, I find it a bit redundant but typeconv keeps it by default. The issue was discussed https://github.com/grantila/typeconv/issues/13, I have asked to see if it's something they would consider adding.
